base/multidimensional.jl

```diff
@@ -510,7 +510,21 @@ module IteratorsMD
 
     # reversed CartesianIndices iteration
 
-    Base.reverse(iter::CartesianIndices) = CartesianIndices(reverse.(iter.indices))
+    @inline function Base._reverse(iter::CartesianIndices, ::Colon)
+        CartesianIndices(reverse.(iter.indices))
+    end
+
+    @inline function Base._reverse(iter::CartesianIndices, dim::Integer)
+        indices = iter.indices
+        len = length(indices)
+        CartesianIndices(Tuple([i == dim ? reverse(indices[i]) : indices[i] for i in 1:len]))
+    end
+
+    @inline function Base._reverse(iter::CartesianIndices, dim::Tuple{Vararg{Integer}})
+        indices = iter.indices
+        len = length(indices)
+        CartesianIndices(Tuple([i in dim ? reverse(indices[i]) : indices[i] for i in 1:len]))
+    end
```

Comment:
```suggestion
    Base.@constprop :aggressive function Base._reverse(iter::CartesianIndices, dim::Integer)
        1 <= dim <= ndims(iter) || throw(ArgumentError(lazy"invalid dimension $dim in reverse"))
        indices = iter.indices
        return CartesianIndices(Base.setindex(indices, reverse(indices[dim]), dim))
    end
    
    Base.@constprop :aggressive function Base._reverse(iter::CartesianIndices, dims::Tuple{Vararg{Integer}})
        indices = iter.indices
        dimrev = ntuple(in(dims), Val(length(indices)))
        length(dims) == sum(dimrev) || throw(ArgumentError(lazy"invalid dimensions $dims in reverse"))
        indices′ = foldl(dims; init = indices) do i, dim
            Base.setindex(i, reverse(i[dim]), dim)
        end 
        return CartesianIndices(indices′)
    end
```
It would be good to make these functions const-propagable.
And the current implement changes the error for invalid `dim` e.g.

Also needs test.